### PR TITLE
tools: Improve final message for release-plugin.sh

### PR DESCRIPTION
**Pull request by anomiex**
## Proposed changes:

Instead of assuming that all plugins except Jetpack have autotagger and wp-svn-autopublish set, look at each plugin's composer.json to output customized messaging.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1718198708233069-slack-C01U2KGS2PQ

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Probably the easiest way to test is to create a file like this:
  ```bash
  #!/bin/bash
  
  # Update this path to whatever is correct for your checkout
  BASE=/usr/local/src/automattic/jetpack

  declare -A PROJECTS
  for ARG in "$@"; do
      PROJECTS["$ARG"]=1
  done
  
  # Copy lines 324+ of tools/release-plugin.sh here
  ```
  Then run the file, passing various combinations of plugins as arguments. For example, `./test.sh plugins/jetpack` should print only the "MANUALBOTH" message, `./test.sh plugins/search` should print only the "AUTO" message, `./test.sh plugins/mu-wpcom-plugin` should print no message (since it auto-tags and isn't in SVN), `./test.sh plugins/jetpack plugins/search` should print both, and so on.

### CODE REVIEW

**Review by jeherve — APPROVED** [review]
That's quite a few use-cases. :)

The changes make sense to me.

File: tools/release-plugin.sh [hunk 1]
```diff
@@ -321,16 +321,91 @@ gh pr create --title "Backport $PLUGINS_CHANGED Changes" --body "$(cat .github/f
 rm .github/files/TEMP_BACKPORT_RELEASE_CHANGES.md
 
 yellow "Release script complete!"
-cat <<"EOM"
 
-Next you need to merge the above PR into trunk.
+echo ''
```
Comment (tbradsha) [inline tools/release-plugin.sh:325]: Any nuanced difference in echoing an empty string vs omitting it?

File: tools/release-plugin.sh [hunk 2]
```diff
@@ -321,16 +321,91 @@ gh pr create --title "Backport $PLUGINS_CHANGED Changes" --body "$(cat .github/f
...
+	EOM
+fi
+
+if [[ ${#MANUALTAG[@]} -gt 0 ]]; then
+	cat <<-EOM
+
+	For these plugins: ${MANUALTAG[*]}
+	Wait for the changes to appear in the mirror repo and conduct a GitHub
+	release. The changes will then be automatically released to SVN and you can
+	then smote test the release. Once ready, use \`./tools/stable-tag.sh <plugin>\`
```
Comment (tbradsha) [inline tools/release-plugin.sh:386]:
```suggestion
	then smoke test the release. Once ready, use \`./tools/stable-tag.sh <plugin>\`
```

**Review by tbradsha — COMMENTED** [review]
Heh, I was reviewing this but it seems it was already approved/merged. Looks good, with one small typo. :^)

File: tools/release-plugin.sh (hunk 1 above)
Comment (anomiex) [reply tools/release-plugin.sh:325]: Not that I know of. It just makes more sense to me this way for some reason.

**Pull request merged**